proxy/proxy_test.go

```diff
@@ -295,6 +295,18 @@ func TestProxyMultipleOrigins(t *testing.T) {
 	defer api.Close()
 
 	unvalidatedIngress := []config.UnvalidatedIngressRule{
+		{
```

Comment:
Can you add two additional tests:
1. An empty regex replacement (regex: "")
2. At least two regex back-reference replacements (will help others see how to use it, in addition to updated docs)

Reply:
1) Do you have any suggestions as to how an empty regex is supposed to work? Right now it reports a configuration error because empty regex doesn't match anything by itself. Should it plainly replace the URL path without back-referencing anything? If so, won't that simply allow for misconfiguration with $'s being sent to services?
2) I don't exactly understand what you meant, but I added a test where /service3/_health/func gets remapped to /func/_health. Is that good enough?

Reply:
1. For the case of the empty regex, I think we want to just skip it. So this would mean that `Path` _must_ exist (and be a valid regex) if you provide a `Rewrite` replacement. If someone wants to provide an empty rewrite from something like `example.com/path` to `example.com` maybe we just recommend that they set the rewrite to `/`? So for the test, just make sure to have a test for `Path: ""` and `Rewrite: "/$1"` (which is expected to fail).
2. Yup, perfect!

Reply:
Such a test is already present and specifying rewrite with an empty path is being handled as an error.
This behavior can be changed to silently omit rewrite, but that doesn't seem right.
One thing that could be done is allowing for `Rewrite` to pass with an empty `Path` in case it doesn't contain back-reference replacements, but I don't know whether it's a good idea or not.
https://github.com/cloudflare/cloudflared/blob/4b63132e770f0e633322fb2319f2f2bd6fbf903c/ingress/ingress.go#L297-L299
https://github.com/cloudflare/cloudflared/blob/4b63132e770f0e633322fb2319f2f2bd6fbf903c/ingress/ingress_test.go#L472-L482

Yes, rewriting `example.com/path` to `example.com` is achieved by rewriting to `/`, I don't know how it could be done otherwise without breaking compatibility for old configs while being more or less intuitive.
Given previous example:
```yaml
  - hostname: subdomain.domain.com
    path: ^/path$
    rewrite: /
    service: https://localhost:3001
```
or if you want to rewrite it as a directory:
```yaml
  - hostname: subdomain.domain.com
    path: ^/path/(.*)$
    rewrite: /$1
    service: https://localhost:3001
```